frontend/src/concepts/k8s/pods/utils.ts

```diff
@@ -14,32 +14,41 @@ export const downloadFullPodLog = async (
   namespace: string,
   podName: string,
   containerName: string,
+  podCompleted: boolean | undefined,
 ) =>
   getPodContainerLogText(namespace, podName, containerName).then((content) =>
-    downloadString(`${podName}-${containerName}.log`, content),
+    downloadString(
+      `${podName}-${containerName}-${podCompleted ? 'completed' : 'incomplete'}.log`,
+      content,
+    ),
   );
 
 export const downloadAllStepLogs = async (
   podContainers: PodContainer[],
   namespace: string,
-  podName: string,
+  pod: PodKind,
 ) => {
   const logPromises = podContainers
     .filter((podContainer) => podContainer !== null)
     .map(async (podContainer) => {
       const logsIndividualStep = await getPodContainerLogText(
         namespace,
-        podName,
+        pod.metadata?.name ?? '',
         podContainer.name,
       );
       return `=============
-${podContainer.name}
+${podContainer.name}  
 =============
 ${logsIndividualStep}`;
     });
-
+  const completed = (pod.status?.containerStatuses || []).every(
+    (containerStatus) => containerStatus?.state?.terminated,
+  );
   const allStepLogs = await Promise.all(logPromises);
 
   const combinedLogs = allStepLogs.join('\n');
-  downloadString(`${podName}.log`, combinedLogs);
+  downloadString(
+    `${pod.metadata?.name}-${completed ? 'completed' : 'incomplete'}.log`,
```

Comment:
I think this will be `complete` instead of `completed`. @yih-wang @kaedward We are currently naming the log file `podname-complete.log` or `podname-incomplete.log` (for all step pod logs) based on the status. Do we want to go ahead with this naming? (For context, this was requested here: https://github.com/opendatahub-io/odh-dashboard/pull/1724#discussion_r1311909409)

Reply:
Honestly, my English isn't the best for making those decisions :P I'm gonna let @kaedward handle it. But the file name format looks good to me.

Reply:
I think `completed` vs `incomplete` might not even be the best way to go. I think we may want a timestamp or something to make this unique per mid-download.

I don't think we need to suffix the log name with completed/complete.

Examples:
```
foo-pod.log // completed
foo-pod-20231116144600.log // 2023 nov 11th @ 2pm (14) 46 minutes 0 seconds
```
This way if they download and download and download it will create a new file with a timestamp value (we can go miliseconds too -- doesn't really need to be readable)

Not having a suffix will make things easier to find.

Thoughts?

Reply:
I'm good with adding the timestamp, but I still think there is a need to add something to denote what is complete and what is not, otherwise, this might look like some random metadata. How will the user know that the incomplete ones have a timestamp attached and the complete ones do not? I had a chance to go check the download actions in OpenShift console, they don't have a timestamp or complete/incomplete.

Reply:
We can say `partial` in the log name to help identify it:
```
foo-pod.log // completed
foo-pod-partial-20231116144600.log // partial log, on 2023 nov 11th @ 2pm (14) 46 minutes 0 seconds
```

Reply:
I leave this to @yih-wang to make the final call.

Reply:
I'm not a huge fan of the suffix... but I'm not against the decision... let me know if you have any further questions @manaswinidas

Reply:
@andrewballantyne ,Can I  use timestamp format like this conditional-2023-11-24T05_29_05.633Z.log instead of YYYYMMDDHHmmss format?

Reply:
@pnaik1 it's more about uniqueness than readable. I just figured YYYYMMDDHHmmss was better than ms from epoch. You could just use `Date().toIsoFormat()` and I imagine that'll be fine too.